src/goto-instrument/accelerate/accelerate.cpp

```diff
@@ -86,7 +86,10 @@ bool acceleratet::contains_nested_loops(goto_programt::targett &loop_header)
   return false;
 }
 
-int acceleratet::accelerate_loop(goto_programt::targett &loop_header)
+int acceleratet::accelerate_loop(
+  goto_programt::targett &loop_header,
+  const double random_seed,
+  const double random_var_freq)
```

Comment:
There is some design problem here: passing in completely unrelated options to some function. At bare minimum pass along `optionst` so that it isn't necessary to add one parameter for each future option.

Reply:
The design problem is that the number of arguments *might* grow in the future. I agree.

But I think that concerns should remain separated. Minisat should not parse command line arguments. I'd humbly suggest passing a reference to typed "config" struct. Which in this case would take only 2 arguments.

Reply:
I largely don't mind whether it's an `optionst` passed around (`optionst` being the result of parsed `cmdlinet` arguments) or actually going for `config`, which is global anyway and would drastically reduce the size of this PR.

Reply:
Globals are not a solution. Since every object (and many functions) requires its own, often large configuration, storing all arguments in a global uncategorized, untyped hashmap would be an architectural disaster.

Globals are not an answer. Better parsing and grouping arguments is.

Reply:
Globals are what we have at the moment.  Whether we should have them or not is a different question to implementing this feature.

Reply:
FWIW: "storing all arguments in a global uncategorized, untyped hashmap would be an architectural disaster." is not what `configt` is/does.